core/appeal/service.go

```diff
@@ -391,125 +385,135 @@ func (s *Service) MakeAction(ctx context.Context, approvalAction domain.Approval
 				return nil, err
 			}
 			continue
-		} else {
-			if approval.Status != domain.ApprovalStatusPending {
-				if err := checkApprovalStatus(approval.Status); err != nil {
-					return nil, err
-				}
-			}
+		}
 
-			if !utils.ContainsString(approval.Approvers, approvalAction.Actor) {
-				return nil, ErrActionForbidden
+		if approval.Status != domain.ApprovalStatusPending {
```

Comment:
moving this code out from the `else` block will change the logic as well. Have we tested this?

Reply:
yes, existing tests are working with high coverages, inside the `if` there only 2 possibilities, either `return` or `continue`, it won't reach next lines, 
so we can remove the else here

https://github.com/odpf/guardian/blob/ae6b057ad98d1efefc00d8b0957747670efacfe7/core/appeal/service.go#L392-L397